include/nbl/video/IGPUObjectFromAssetConverter.h

```diff
@@ -591,76 +710,189 @@ auto IGPUObjectFromAssetConverter::create(const asset::ICPUMesh** const _begin,
-                    tmpViewParams.subresourceRange.layerCount = params.arrayLayers;
-                    auto tmpView = m_driver->createGPUImageView(std::move(tmpViewParams));
-                    // deprecated OpenGL path (do with compute shader in the future)
-                    tmpView->regenerateMipMapLevels();
-                }
+                IGPUCommandBuffer::SImageMemoryBarrier imgbarrier;
+                // TODO layouts
+                imgbarrier.oldLayout = asset::EIL_UNDEFINED;
+                imgbarrier.newLayout = asset::EIL_UNDEFINED;
+                imgbarrier.srcQueueFamilyIndex = _params.transferQueue->getFamilyIndex();
+                imgbarrier.dstQueueFamilyIndex = _params.computeQueue->getFamilyIndex();
+                imgbarrier.image = gpuimg;
+                asset::IImage::SSubresourceRange subres;
+                subres.baseArrayLayer = 0u;
+                subres.layerCount = params.arrayLayers;
+                subres.baseMipLevel = 0u;
+                subres.levelCount = params.mipLevels;
+                imgbarrier.subresourceRange = subres;
+                asset::SMemoryBarrier membarrier;
+                membarrier.srcAccessMask = asset::EAF_TRANSFER_WRITE_BIT;
+                membarrier.dstAccessMask = asset::EAF_SHADER_READ_BIT;
+                imgbarrier.barrier = membarrier;
+
+                imgMemBarriers.push_back(imgbarrier);
             }
+
+            auto gpubuf_found = img2gpubuf.find(cpuimg);
+            auto& gpubuf = gpubuf_found->second;
+
+            // TODO layout
+            cmdbuf->copyBufferToImage(gpubuf.get(), gpuimg.get(), asset::EIL_UNDEFINED, count, regions.begin());
 		}
 
 		res->operator[](i) = std::move(gpuimg);
     }
 
+    if (!cmdbuf)
+        return res;
+
+    // for when mipmaps fill be generated with compute shader
+    cmdbuf->pipelineBarrier(asset::EPSF_TRANSFER_BIT, asset::EPSF_COMPUTE_SHADER_BIT, 0u, 0u, nullptr, 0u, nullptr, imgMemBarriers.size(), imgMemBarriers.data());
+    cmdbuf->end();
+
+    auto imgsFilledSemaphore = _params.device->createSemaphore();
+    auto* imgsFilledSemaphore_ptr = imgsFilledSemaphore.get();
+
+    IGPUQueue::SSubmitInfo submit;
+    submit.commandBufferCount = 1u;
+    auto* cmdbuf_ptr = cmdbuf.get();
+    submit.commandBuffers = &cmdbuf_ptr;
+    submit.signalSemaphoreCount = 1u;
+    submit.pSignalSemaphores = &imgsFilledSemaphore_ptr;
+    submit.waitSemaphoreCount = 0u;
+    submit.pWaitDstStageMask = nullptr;
+    submit.pWaitSemaphores = nullptr;
+    // dont event tell the queue to signal a fence after last submit if it'll never be touched by user
+    auto* signalFence = _params.transferFence ? fence.get() : nullptr;
+    _params.transferQueue->submit(1u, nullptr, signalFence);
+    if (_params.transferFence && fence)
+        _params.transferFence[0] = std::move(fence);
+    if (_params.transferSemaphore)
+        _params.transferSemaphore[0] = imgsFilledSemaphore;
```

Comment:
instead of having to do this in every overload of a `create` call, the top level call (`getGPUObjectsFromAssets_impl` or whatever it was called) should be expected to:
1. If a non-null event pointer has been provided, record a final event signal into the commandbuffer
2. End the commandbuffer
3. Submit the commandbuffer with the appropriate sempahores and fences